packages/vite/src/node/shortcuts.ts

```diff
@@ -44,8 +44,11 @@ export function bindShortcuts(
   const onInput = async (input: string) => {
     // ctrl+c or ctrl+d
     if (input === '\x03' || input === '\x04') {
-      process.emit('SIGTERM')
-      return
+      try {
+        await server.close()
+      } finally {
+        process.exit(1)
+      }
```

Comment:
After multiple tests, I think this the best fix, because it keeps the default behaviour of ctrl+c
```suggestion
      process.stdin.setRawMode(false)
      process.stdin.write(input)
      return
```